libsolidity/ast/ASTUtils.cpp

```diff
@@ -19,12 +19,32 @@
 #include <libsolidity/ast/AST.h>
 #include <libsolidity/ast/ASTUtils.h>
 
+#include <libsolutil/Algorithms.h>
+
 namespace solidity::frontend
 {
 
+bool isConstantVariableRecursive(VariableDeclaration const& _varDecl)
+{
+	solAssert(_varDecl.isConstant(), "Constant variable expected");
+	auto visitor = [&](VariableDeclaration const& _variable, util::CycleDetector<VariableDeclaration>& _cycleDetector, size_t _depth)
+	{
+		solAssert(_depth < 256, "Recursion depth limit reached");
+
+		if (auto identifier = dynamic_cast<Identifier const*>(_variable.value().get()))
```

Comment:
If you are only looking for `identifier`, then this cannot catch the circularity as in the example in https://github.com/ethereum/solidity/pull/10783/files#r560848021 does it?

Reply:
I added a test for it. https://github.com/ethereum/solidity/pull/10783/files#diff-31b40ad134e36fb598eba9d52a0d3ac9d7f983fbdd03793ade820316a8b4743e 

It doesn't detect the circular part though. Let me see if this function can be improved to deal with that case.

Reply:
Fixed this now.